2017/proposals/bansal-utkarsh.md

```diff
@@ -0,0 +1,628 @@
+ found [here](http://doc.pytest.org/en/latest/writing_plugins.html) .
+    * Capture error (`capture_err.py`) This plugin captures stderr during test execution. If the test fails
+    or raises an error, the captured output is appended to the error or failure output. Pytest already has this
+feature inbuilt so this plugin can be dropped.
+
+        **Example**
+        ```python
+        def check_even(n, nn):
+            print(n)
+            assert n % 2 == 0 or nn % 2 == 0
+
+
+        def test_evens():
+            for i in range(0, 5):
+                yield check_even, i, i * 3
+        ```
+
+        **Output**
+        ```
+        ======================================= FAILURES =======================================
+        ____________________________________ test_evens[1] _____________________________________
+
+        n = 1, nn = 3
+
+            def check_even(n, nn):
+                print(n)
+        >       assert n % 2 == 0 or nn % 2 == 0
+        E       assert ((1 % 2) == 0 or (3 % 2) == 0)
+
+        experiments.py:37: AssertionError
+        --------------------------------- Captured stdout call ---------------------------------
+        1
+        ____________________________________ test_evens[3] _____________________________________
+
+        n = 3, nn = 9
+
+            def check_even(n, nn):
+                print(n)
+        >       assert n % 2 == 0 or nn % 2 == 0
+        E       assert ((3 % 2) == 0 or (9 % 2) == 0)
+
+        experiments.py:37: AssertionError
+        --------------------------------- Captured stdout call ---------------------------------
+        3
+        ```
+
+
+    * Cleanup (`cleanup.py`) This plugin deletes XDR offset files after all testing is done. This will have to
+    be re-written as a pytest plugin.
+    * Knownfailure (`knownfailure.py`) This is a decorator to check tests that are expected to fail. Pytest has
+    something similar to this called `xfail`. The documentation for this is available [here](http://doc.pytest.org/en/latest/skipping.html#mark-a-test-function-as-expected-to-fail) .
+    This needs discussion to finalize if we need to port the existing code or if `xfail` is good enough for our use.
+    * Memory leak (`memleak.py`) This plugin detects memory leaks in tests. There is an open-source (MIT license) plugin, [pytest-leaks](https://github.com/abalkin/pytest-leaks).
+    This also needs discussion to finalize if we need to port the existing code or if pytest-leaks is good enough for our use.
+    * Open files (`open_files.py`) This plugin lists all open files when a test fails or errors. They also are listed at
+    the end of the test suite. This plugin is currently disabled because we were facing issue with capturing stdout. Discussion
+    is needed to decide if we need this plugin anymore. If we do, this will have to be rewritten for pytest as no open source
+    alternatives are available.
```

Comment:
Yes we really need that plugin. But it doesn't need to be part of the first PR. I assume it can be implemented like the simple memory leak finder.